packages/server/src/adapters/node-http/content-type/form-data/index.ts

```diff
@@ -0,0 +1,66 @@
+import busboy, { BusboyHeaders } from '@fastify/busboy';
+import { createNodeHTTPContentTypeHandler } from '../../internals/contentType';
+
+export const nodeHTTPFormDataContentTypeHandler =
+  createNodeHTTPContentTypeHandler({
+    isMatch(opts) {
+      const contentTypeHeader = opts.req.headers['content-type'];
+
+      return (
+        contentTypeHeader?.startsWith('multipart/form-data') ||
+        contentTypeHeader === 'application/x-www-form-urlencoded'
+      );
+    },
+    async getBody(opts) {
+      const { req } = opts;
+
+      if (req.method === 'GET') {
+        const form = new FormData();
+
+        for (const [key, value] of Object.entries(req.query ?? {})) {
+          form.append(key, value as any);
+        }
+
+        return { ok: true, data: form };
+      }
+
+      const bb = busboy({ headers: req.headers as BusboyHeaders });
+      const form = new FormData();
+
+      await new Promise((resolve, reject) => {
+        bb.on('file', async (name, file, filename, _, mimeType) => {
+          const chunks: any[] = [];
+          await new Promise((resolve, reject) => {
+            file.on('data', (chunk) => {
+              chunks.push(chunk);
+            });
+            file.on('error', reject);
+            file.on('end', resolve);
+          });
+
+          const buffer = Buffer.concat(chunks);
+
+          form.append(
+            name,
+            new File([buffer], filename, { type: mimeType }) as Blob,
+          );
+        });
+
+        bb.on('field', (name, value) => {
+          form.append(name, value);
+        });
+
+        bb.on('error', reject);
+        bb.on('finish', resolve);
+
+        req.pipe(bb);
```

Comment:
Does this mean that all the files and their data must be received before reaching the actual resolver?

Not sure if it's the case cause I haven't played with it, but I feel like the files could/should be streamed into the procedure so we can for instance stream big files straight to disk or to s3 etc and not held in memory

Reply:
good catch, it's being fully read into memory right now

Reply:
Is there a way to forward the stream instead?

Reply:
The current setup work fine for small files but will eat the computer alive for any big ones